transform-plugins/src/main/java/co/cask/hydrator/plugin/ValueMapper.java

```diff
@@ -0,0 +1,228 @@
+    StructuredRecord.Builder builder = createOutputRecords(input, outputSchema);
+    StructuredRecord ss = builder.build();
+    LOG.debug("in transform end :" + ss.get("designationName"));
+    emitter.emit(ss);
+  }
+
+  /**
+   * Creates output structure record using mapping dataset and set them to builder.
+   */
+  private StructuredRecord.Builder createOutputRecords(StructuredRecord input, Schema outputSchema) {
+
+    StructuredRecord.Builder builder = StructuredRecord.builder(outputSchema);
+    for (Schema.Field sourceField : input.getSchema().getFields()) {
+
+      LOG.debug("in createOutputRecords :" + input.get(sourceField.getName()
+      ).toString());
+      if (!input.get(sourceField.getName()).equals("NULL") &&
+              !input.get(sourceField.getName()).equals("EMPTY") &&
+              fieldsMapping.containsKey(sourceField.getName())) {
+        builder = lookUpAndSetValues(input, sourceField, builder);
+        LOG.debug("in createOutputRecords after :" + builder);
+      } else {
+        // for those source field whose values are either NULL or EMPTY
+        builder.set(sourceField.getName(), "" + input.get(sourceField.getName()
+        ).toString());
+      }
+    }
+    LOG.debug("in createOutputRecords end :" + builder);
+    return builder;
+  }
+
+  /**
+   * Sets target fields and values from mapping dataset
+   */
+  private StructuredRecord.Builder lookUpAndSetValues(StructuredRecord input, Schema.Field sourceField,
+                                                      StructuredRecord.Builder builder) {
+    if (datasetMapping.containsKey(sourceField.getName())) {
+      ValueMapperLookUp valueMapperLookUp = datasetMapping.get(sourceField.getName());
+      LOG.debug("in lookUpAndSetValues:" + sourceField.getName());
+
+      if (valueMapperLookUp.lookup(input.get(sourceField.getName()).toString()) != null &&
+              valueMapperLookUp.lookup(input.get(sourceField.getName()).toString()).toString().trim().length() > 0) {
+        LOG.debug("in lookUpAndSetValues lookup value:" + valueMapperLookUp.lookup(input.get(sourceField
+                .getName
+                        ())
+                .toString()));
+        builder.set(fieldsMapping.get(sourceField.getName()),
+                valueMapperLookUp.lookup(input.get(sourceField.getName()).toString()));
+      } else {
+        builder.set(fieldsMapping.get(sourceField.getName()), "Empty");
+      }
+    } else {
+      // for those source field whose mapping is not present
+      builder.set(fieldsMapping.get(sourceField.getName()), "Empty");
+    }
+
+    return builder;
+  }
+
+  /**
+   * Creates output schema record using source and destination field mapping provided by user.
+   */
+  private Schema getOutputSchema(Schema inputSchema) throws IllegalArgumentException {
+    Schema outputSchema = schemaCache.get(inputSchema);
+    if (outputSchema != null) {
+      return outputSchema;
```

Comment:
It is expected behaviour. If the object is null, it is initialized otherwise it returns the same object. No need to throw or handle the exception. Please confirm if the understanding is correct.